packages/payment-detection/src/thegraph/queries/GetPaymentsAndEscrowState.graphql

```diff
@@ -27,15 +27,76 @@ fragment EscrowEventResult on EscrowEvent {
   gasPrice
 }
 
-query GetPaymentsAndEscrowState($reference: Bytes!, $to: Bytes!) {
-  payments(where: { reference: $reference, to: $to }, orderBy: timestamp, orderDirection: asc) {
+query GetPaymentsAndEscrowState(
+  $reference: Bytes!
+  $to: Bytes!
+  $tokenAddress: Bytes
+  $contractAddress: Bytes!
+) {
+  payments(
+    where: {
+      reference: $reference
+      to: $to
+      contractAddress: $contractAddress
+      currency: null
+      tokenAddress: $tokenAddress
```

Comment:
This is not related but similar, let's filter everything we can at the same place.